app/Jobs/ProcessNewVolumeFiles.php

```diff
@@ -60,10 +72,52 @@ public function handle()
             ->when($this->only, fn ($query) => $query->whereIn('id', $this->only));
 
         if ($this->volume->isImageVolume()) {
-            $query->eachById([ProcessNewImage::class, 'dispatch']);
+            $query->eachById(function (Image $img) {
+                // Process new image, if volume is just created
+                if (!$this->uuidMap) {
+                    ProcessNewImage::dispatch($img);
+                    return;
+                }
+
+                $prefix = fragment_uuid_path($this->uuidMap[$img->uuid]);
+                $copyPrefix = fragment_uuid_path($img->uuid);
+
+                $format = config('thumbnails.format');
+                $hasThumbnail = Storage::disk(config('thumbnails.storage_disk'))->exists($prefix.".{$format}");
+                $hasTiledData = count(Storage::disk(config('image.tiles.disk'))->files($prefix)) > 0;
+                
+                if ($hasThumbnail && $hasTiledData) {
+                    CloneImageThumbnails::dispatch($prefix, $copyPrefix);
+                } else {
+                    ProcessNewImage::dispatch($img);
+                }
```

Comment:
Checking for file existence for all files in a single job will take a very long time. It's better to keep the runtime of individual jobs low and submit more jobs instead. So here you should check for file existence in CloneImageThumbnails/CloneVideoThumbnails. If the files do not exist, call ProcessNewImage/ProcessNewVideo from there.

For this you probably have to pass the full oldImg and newImg models to the CloneThumbnails jobs (instead of just the UUIDs) but that's ok.